gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1612,14 +1616,194 @@ gst_tensor_src_iio_create (GstBaseSrc * src, guint64 offset,
+ * @param[in/out] buffer_map Gst buffer map to write data to
+ *
+ * assumes each data starting point is byte aligned
+ */
+static void
+gst_tensor_src_iio_process_scanned_data (GList * channels, gchar * data,
+    gfloat * buffer_map)
+{
+  GList *list;
+  GstTensorSrcIIOChannelProperties *prop;
+  guint64 storage_mask;
+  guint channel_idx;
+
+  for (list = channels, channel_idx = 0; list != NULL;
+      list = list->next, channel_idx += 1) {
+    prop = list->data;
+    switch (prop->storage_bytes) {
+      case 1:
+      {
+        guint8 value = *(guint8 *) (data + prop->location);
+        /** right shift the extra storage bits */
+        value >>= (8 - prop->storage_bits);
+        buffer_map[channel_idx] =
+            gst_tensor_src_iio_process_scanned_data_from_guint8 (prop, value);
+        break;
+      }
+      case 2:
+      {
+        guint16 value = *(guint16 *) (data + prop->location);
+        if (prop->big_endian) {
+          value = be16toh (value);
+          /** right shift the extra storage bits for big endian */
+          value >>= (16 - prop->storage_bits);
+        } else {
+          value = le16toh (value);
+          /** mask out the extra storage bits for little endian */
+          storage_mask = (G_GUINT64_CONSTANT (1) << prop->storage_bits) - 1;
+          value &= storage_mask;
+        }
+        buffer_map[channel_idx] =
+            gst_tensor_src_iio_process_scanned_data_from_guint16 (prop, value);
+        break;
+      }
+      case 3:
+      /** follow through */
+      case 4:
+      {
+        guint32 value = *(guint32 *) (data + prop->location);
+        if (prop->big_endian) {
+          value = be32toh (value);
+          /** right shift the extra storage bits for big endian */
+          value >>= (32 - prop->storage_bits);
+        } else {
+          value = le32toh (value);
+          /** mask out the extra storage bits for little endian */
+          storage_mask = (G_GUINT64_CONSTANT (1) << prop->storage_bits) - 1;
+          value &= storage_mask;
+        }
+        buffer_map[channel_idx] =
+            gst_tensor_src_iio_process_scanned_data_from_guint32 (prop, value);
+        break;
+      }
+      case 5:
+      /** follow through */
+      case 6:
+      /** follow through */
+      case 7:
+      /** follow through */
+      case 8:
+      {
+        guint64 value = *(guint64 *) (data + prop->location);
+        if (prop->big_endian) {
+          value = be64toh (value);
+          /** right shift the extra storage bits for big endian */
+          value >>= (64 - prop->storage_bits);
+        } else {
+          value = le64toh (value);
+          /** mask out the extra storage bits for little endian */
+          storage_mask = (G_GUINT64_CONSTANT (1) << prop->storage_bits) - 1;
+          value &= storage_mask;
+        }
+        buffer_map[channel_idx] =
+            gst_tensor_src_iio_process_scanned_data_from_guint64 (prop, value);
+        break;
+      }
```

Comment:
Handle errors with ```default:```

Reply:
There is a `g_return_val_if_fail (prop->storage_bytes <= 8, FALSE)` on line 559 when reading `storage_bytes`
This function is called after correctly reading `storage_bytes`
Added default section for completeness

Reply:
Oh. ok then.